homeassistant/components/rainforest_raven/sensor.py

```diff
@@ -0,0 +1,204 @@
+        message_key="CurrentSummationDelivered",
+        key="summation_delivered",
+        name="Total Meter Energy Delivered",
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        device_class=SensorDeviceClass.ENERGY,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+    ),
+    RAVEnSensorEntityDescription(
+        message_key="CurrentSummationDelivered",
+        key="summation_received",
+        name="Total Meter Energy Received",
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        device_class=SensorDeviceClass.ENERGY,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+    ),
+    RAVEnSensorEntityDescription(
+        message_key="InstantaneousDemand",
+        key="demand",
+        name="Meter Power Demand",
+        native_unit_of_measurement=UnitOfPower.KILO_WATT,
+        device_class=SensorDeviceClass.POWER,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+)
+
+
+DIAGNOSTICS = (
+    RAVEnSensorEntityDescription(
+        message_key="NetworkInfo",
+        key="link_strength",
+        name="Link Strength",
+        native_unit_of_measurement=PERCENTAGE,
+        state_class=SensorStateClass.MEASUREMENT,
+        device_class=SensorDeviceClass.SIGNAL_STRENGTH,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        attribute_keys={
+            "Channel": "channel",
+        },
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up a config entry."""
+    coordinator = hass.data[DOMAIN][entry.entry_id]
+    entities: list[RAVEnSensor] = [
+        RAVEnSensor(coordinator, description) for description in DIAGNOSTICS
+    ]
+
+    for meter_mac_addr in entry.data[CONF_MAC]:
+        entities.extend(
+            RAVEnMeterSensor(coordinator, description, meter_mac_addr)
+            for description in SENSORS
+        )
+
+        meter_data = coordinator.data.get("Meters", {}).get(meter_mac_addr) or {}
+        if meter_data.get("PriceCluster", {}).get("currency"):
+            entities.append(
+                RAVEnMeterSensor(
+                    coordinator,
+                    RAVEnSensorEntityDescription(
+                        message_key="PriceCluster",
+                        key="price",
+                        name="Meter Price",
+                        native_unit_of_measurement=f"{meter_data['PriceCluster']['currency'].value}/{UnitOfEnergy.KILO_WATT_HOUR}",
+                        device_class=SensorDeviceClass.MONETARY,
+                        state_class=SensorStateClass.MEASUREMENT,
+                        attribute_keys={
+                            "Tier": "tier",
+                            "Rate": "rate_label",
+                        },
+                    ),
+                    meter_mac_addr,
+                )
+            )
+
+    async_add_entities(entities)
+
+
+class RAVEnSensor(CoordinatorEntity, SensorEntity):
```

Comment:
```suggestion
class RAVEnSensor(CoordinatorEntity[RAVEnDataCoordinator], SensorEntity):
```

For new integrations, we require the following:
- Support for [the new entity naming](https://developers.home-assistant.io/docs/core/entity#has_entity_name-true-mandatory-for-new-integrations)
- Services must be [translatable](https://developers.home-assistant.io/blog/2023/07/11/translating-services)
- Entities and their attributes should be [translatable](https://developers.home-assistant.io/blog/2023/03/27/entity_name_translations/)

Please adopt your code to add support for it.

Reply:
Added translations in d4e82a9. Still working on the typing change here to `CoordinatorEntity[RAVEnDataCoordinator]`.

Reply:
Alright, the typing change is done now too.